dipy/viz/horizon/util.py

```diff
@@ -63,3 +63,32 @@ def check_img_dtype(images):
             warnings.warn(msg.format(idx + 1))
 
     return valid_images
+
+
+def is_binary_image(data, unique_points=100):
+    """Check if an image is binary image.
+
+    Parameters
+    ----------
+    data : ndarray
+    unique_points : int, optional
+        number of points to sample the check, by default 100
+
+    Returns
+    -------
+    boolean
+        Whether the image is binary or not
+    """
+    indices = []
+
+    for dim in data.shape:
+        indices.append(
+            np.random.choice(
+                np.arange(0, dim - 1),
+                replace=dim <= unique_points,
+                size=(unique_points)
+                )
+            )
+
+    indices = np.asarray(indices)
+    return np.unique(data[*indices]).shape[0] <= 2
```

Comment:
Not sure to understand the goal of this PR since you are still using `np.unique` here. I also do not understand the initial request.

Reply:
He is simply checking if an image is binary. Now only running unique for 100 values. Not the entire image.

Reply:
why not `np.histogram`?

Reply:
ok, for now

Reply:
@skoudoro np.histogram will try to scan the entire data which is almost equal to checking np.unique, here we are simply generating 100 indices for the data and testing them.